libdnf/rpm/solv/goal_private.cpp

```diff
@@ -396,23 +396,25 @@ libdnf::solv::IdQueue GoalPrivate::list_obsoleted() {
     return list_results(SOLVER_TRANSACTION_OBSOLETED, 0);
 }
 
-void GoalPrivate::write_debugdata(const std::string & dir) {
+void GoalPrivate::write_debugdata(const std::filesystem::path & dir) {
     libdnf_assert_goal_resolved();
 
+    // Ensures the presence of the directory.
+    auto dir_path = std::filesystem::absolute(dir);
+    std::filesystem::create_directories(dir_path);
+
+    // Removes old debug data, if it exists.
+    for (const auto & dir_entry : std::filesystem::directory_iterator(dir_path)) {
+        std::filesystem::remove(dir_entry);
+    }
+
     int flags = TESTCASE_RESULT_TRANSACTION | TESTCASE_RESULT_PROBLEMS;
-    // TODO(jmracek) add support of relative path and create required dirs and parents
-    //     g_autofree char * absdir = abspath(dir);
-    //     if (!absdir) {
-    //         std::string msg = tfm::format(_("failed to make %s absolute"), dir);
-    //         throw Goal::Error(msg, DNF_ERROR_FILE_INVALID);
-    //     }
-    //     makeDirPath(dir);
-    //     g_debug("writing solver debugdata to %s", absdir);
-    auto ret = ::testcase_write(libsolv_solver, dir.c_str(), flags, NULL, NULL);
+    //g_debug("writing solver debugdata to %s", absdir);
```

Comment:
I believe that `//g_debug("writing solver debugdata to %s", absdir);` can be removed when it will be added to goal event log. Or moved to `[libdnf/base/goal.cpp](https://github.com/rpm-software-management/libdnf/pull/1465/files#diff-2dc6ad27d868e29d832948d260b62e2e2628afdb4d9c19b7e7711359cd65e6b5)` after `p_impl->rpm_goal.write_debugdata(std::filesystem::path(cfg_main.debugdir().get_value()) / "packages");`

Reply:
Why would you put this into the event log? This is not an event that is related to the content of the transaction. The `g_debug()` added a line into the debug log, I'm not sure how useful that log line is, but that's where it should go if we want it.

Reply:
> Why would you put this into the event log?

Probably as information for higher level (caller) that debug solver data was written and where. But on the other hand, caller can assume that they were written if the writing is enabled by configuration and we did not get an error during writting debug data.

Reply:
I suggest that to write it into event log is ok, because the process (writing debug data) is completely handled by configuration (enablement and path) and also it has negative impact on performance (from 0.5 s for standard resolvement to 8s). The message also contains viable information about the path.

Reply:
But it is not a transaction event, hence transaction event log is a wrong place to put it in...